x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -356,6 +410,13 @@ const prepareDefinitionSectionListItems = (
         description: <Filters filters={savedQuery.attributes.filters as Filter[]} />,
```

Comment:
I really had to resort to typecasting here because filters are not typed properly (array of unknowns) and can't be easily typed, it seems: https://github.com/elastic/kibana/blob/97d26c1b4c7bed88d9725ac80e1d22ed1cabf0e9/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.ts#L104C1-L104C1

In the previous, pre-refactoring version the type was also cast:
```typescript
...
  queryBar: {
    query: { query: rule.query ?? '', language: rule.language ?? '' },
    filters: (rule.filters ?? []) as Filter[],
    saved_id: rule.saved_id ?? null,
  },
...
```

Looks like we need some deeper RuleResponse / Rule refactoring to type this properly.